fix: add info about non blocking behavior (#678)
aerogear/mobile-docs: modules/ROOT/pages/_partials/sync/js-client-offline-and-conflict.adoc

For more information, see link:https://www.apollographql.com/docs/react/essentials/mutations.html#props[Apollo's document about mutations].

== Offline Workflow

> When offline `client.mutate` function will return immediately after is called.

Reviewer: Proposal: "When the {sync-client} changes to an offline state, the `client.mutate` function is immediately triggered and  results in an error message being captured. Developers can use this message to determine what the issue is, and write code to handle this situation on the server."

Author: I have tried to apply this but there are some technical problems in the sentence. How about:

`When the mutation is executed while the application is in the ofline state, the client.mutate function **will return** immediately and *return* Promise with an error. Developers can use this *error* object to determine if the error is related to offline state and monitor when a change will be synced back to server`

Reviewer 2: This is better, but would like to rewrite (this is important for offline,right?):

If a mutation occurs while the device is offline, the client.mutate function:
* returns immediately
* returns a promise with an error
You can use check error objects to isolate errors relating to offline state and sync the change back to the server, when the device is online again.

Kinda begs for an example, right? maybe a link to another section or showcase app

Author: @finp WOW. Really really like it! Doing chores now.

Author: Forgot to mention. An example is in code straight below these sentences.

Returned promise will resolve into error (`catch` method is triggered).
Developers can detect if error is an offline error and watch for change to be replicated back to server.

For example: 
[source, javascript]
----
  client.mutate(...).catch((error)=> {
    // 1. Detect if this was an offline error
   if(error.networkError && error.networkError.offline){
     const offlineError: OfflineError =  error.networkError;
     // 2. We can still track when offline change is going to be replicated.
     offlineError.watchOfflineChange().then(...)
   }
  });
----

> NOTE:  In addition to watching individual mutations, you can add a global offline listener when creating a client.

Reviewer 2: is that covered in docs?

Author: Yes. In `== Listening for Events`

Author: Yes in `== Listening for Events`

> Note: Additionally to watching individual mutations framework offers global offline listener that can be supplied when creating client.

Reviewer 2: ```suggestion
NOTE:  In addition to watching individual mutations, you can add a global offline listener when creating a client.
```
do we provide docs for that?

Author: Yes. It `Listening to changes` chapter bellow

For more information, see link:https://www.apollographql.com/docs/react/essentials/mutations.html#props[Apollo's document about mutations].

== Offline Workflow

If a mutation occurs while the device is offline, the client.mutate function:

- returns immediately
- returns a promise with an error

> You can check error object to isolate errors relating to offline state.

Reviewer: ```suggestion
You can check the _error_ object to isolate errors relating to offline states.
```

> `watchOfflineChange` method on error object will allow to watch when a change will be synced back to server,

Reviewer: ```suggestion
To sync an offline change with the server when a device comes back online, use the `watchOfflineChange()` method.
```

Author: This method do not sync the changes. It will watch and notify users when sync will happen.

Reviewer: How does this read:
"Invoking the `watchOfflineChange()` method on an _error_ object watches for when an offline change is synced with the server, and sends an notification when triggered."
